components/camel-salesforce/camel-salesforce-component/src/main/java/org/apache/camel/component/salesforce/internal/processor/JsonRestProcessor.java

```diff
@@ -104,6 +108,11 @@ protected void processRequest(Exchange exchange) throws SalesforceException {
             });
             break;
 
+        case RECENT:
+            // handle known response type
```

Comment:
The `TypeReference` is needed by the Jackson to properly support generics. I think that it should be the other way around (change [JsonRestProcessor.java#L71)](https://github.com/zregvart/camel/blob/9a45774db503967bbfabb0bdad0f63ea8f7ace69/components/camel-salesforce/camel-salesforce-component/src/main/java/org/apache/camel/component/salesforce/internal/processor/JsonRestProcessor.java#L71) that is to inline `TypeReference<List<Version>> `, each time that `case` branch is hit a new object is created only to pass on the type information. From performance standpoint it would be better if that line was inlined as constant. I can do a pull request against that line, didn't want to mix two issues in one PR.